examples/chain/40_OffchainSpotVaults/create_vault/example.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"time"
+
+	"cosmossdk.io/math"
+	"github.com/InjectiveLabs/sdk-go/client/common"
+	"github.com/cosmos/cosmos-sdk/types/tx"
+
+	rpchttp "github.com/cometbft/cometbft/rpc/client/http"
+
+	"github.com/CosmWasm/wasmd/x/wasm/types"
+	chainclient "github.com/InjectiveLabs/sdk-go/client/chain"
+	cosmtypes "github.com/cosmos/cosmos-sdk/types"
+)
+
+func main() {
+	// network := common.LoadNetwork("mainnet", "k8s")
+	network := common.LoadNetwork("testnet", "k8s")
+	tmRPC, err := rpchttp.New(network.TmEndpoint, "/websocket")
+
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	senderAddress, cosmosKeyring, err := chainclient.InitCosmosKeyring(
+		os.Getenv("HOME")+"/.injectived",
+		"injectived",
+		"file",
+		"inj-user",
+		"12345678",
+		"5d386fbdbf11f1141010f81a46b40f94887367562bd33b452bbaa6ce1cd1381e", // keyring will be used if pk not provided
+		false,
+	)
+
+	if err != nil {
+		panic(err)
+	}
+
+	clientCtx, err := chainclient.NewClientContext(
+		network.ChainId,
+		senderAddress.String(),
+		cosmosKeyring,
+	)
+
+	if err != nil {
+		panic(err)
+	}
+
+	clientCtx = clientCtx.WithNodeURI(network.TmEndpoint).WithClient(tmRPC)
+
+	chainClient, err := chainclient.NewChainClient(
+		clientCtx,
+		network.ChainGrpcEndpoint,
+		common.OptionTLSCert(network.ChainTlsCert),
+		common.OptionGasPrices("500000000inj"),
+	)
```

Comment:
`NewChainClient` now should receive the network instance instead of only the endpoint. And the certs are taken from the network directly inside the ChainClient.
```suggestion
	chainClient, err := chainclient.NewChainClient(
		clientCtx,
		network,
		common.OptionGasPrices("500000000inj"),
	)
```